### PR TITLE
set GCC 9.4.0 as default compiler for 2021.12 compat layer

### PR DESCRIPTION
**Pull request by boegel**
cfr. https://github.com/EESSI/software-layer/issues/151

requires https://github.com/EESSI/gentoo-overlay/pull/72

### CODE REVIEW

File: ansible/playbooks/roles/compatibility_layer/tasks/prefix_configuration.yml [hunk 1]
```diff
@@ -10,6 +10,11 @@
...
 
-- name: Make the most recent GCC version the default
-  command: "{{ gentoo_prefix_path }}/usr/bin/gcc-config latest"
+# Set default compiler in compat layer.
+# We should not use the latest version of GCC here, since that will probably cause problems
+# when building older GCC versions with EasyBuild in the software layer.
+# (see https://github.com/EESSI/software-layer/issues/151).
+# Note: the selected GCC version must be installed (cfr. EESSI set in gentoo-overlay)
+- name: Make specific GCC version the default compiler
+  command: "{{ gentoo_prefix_path }}/usr/bin/gcc-config 9.4.0"
```
Comment (bedroge) [inline ansible/playbooks/roles/compatibility_layer/tasks/prefix_configuration.yml:19]: I would use a variable (e.g. `{{ prefix_default_gcc }}`) here, and add that to `defaults/main.yml`? Makes it easier to change this later on.

Comment (boegel) [reply ansible/playbooks/roles/compatibility_layer/tasks/prefix_configuration.yml]: fixed in 8aa6884

**Comment by boegel** [comment]
@bedroge Works like a charm now:

```
ec2-user@ip-172-31-25-60 ~/compatibility-layer/ansible/playbooks $ gcc-config -l
 [1] aarch64-unknown-linux-gnu-9.4.0 *
 [2] aarch64-unknown-linux-gnu-11.2.0
ec2-user@ip-172-31-25-60 ~/compatibility-layer/ansible/playbooks $ gcc --version
gcc (Gentoo 9.4.0 p1) 9.4.0
```

**Comment by bedroge** [comment]
Could you also add an entry for this new variable in `ansible/playbooks/README.md`?

**Comment by boegel** [comment]
> Could you also add an entry for this new variable in `ansible/playbooks/README.md`?

done in cca2b63

**Review by bedroge — APPROVED** [review]
lgtm!

**Pull request merged**